src/main/java/com/example/universe/simulator/common/config/CachingConfig.java

```diff
@@ -0,0 +1,26 @@
+package com.example.universe.simulator.common.config;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.cache.RedisCacheConfiguration;
+import org.springframework.data.redis.serializer.RedisSerializationContext;
+import org.springframework.data.redis.serializer.RedisSerializer;
+
+import java.time.Duration;
+
+@Configuration
+@ConditionalOnProperty(prefix = "app.caching", name = "enabled", havingValue = "true")
+@EnableCaching
+public class CachingConfig {
+
+    @Bean
+    public RedisCacheConfiguration cacheConfiguration(@Value("${spring.cache.redis.time-to-live}") Long cacheTtlSeconds) {
```

Comment:
`spring.cache.redis.time-to-live` property type is `Duration`. You can read about duration conversion [here](https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config.typesafe-configuration-properties.conversion.durations).

Action points:
1. change `Long cacheTtlSeconds` to `Duration`
2. rename `cacheTtlSeconds` to `timeToLive`
3. change the `entryTtl(Duration.ofSeconds(cacheTtlSeconds))` line accordingly
4. update `application.properties` in entity-service
5. update `CachingConfigSmokeTest` test in entity-service

Reply:
Changed.